### PR TITLE
Add autoPad resize action

### PR DESCRIPTION
**Pull request by magdakwiecien**
#### What does this PR solve?
- Add autoPad resize action

#### Final checklist
- [x] Implementation is aligned to Spec.
- [x] Tests - Add proper tests to the added code.

### CODE REVIEW

File: __TESTS__/unit/actions/Resize/AutoPadResizeAction.test.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import {Transformation} from "../../../../src";
+import {Resize, autoPad} from "../../../../src/actions/resize";
+import {Background} from "../../../../src/qualifiers";
+
+describe('Tests for Transformation Action -- Resize.autoPad', () => {
+  it('Ensures it generates the right transformation', () => {
+    const tx = new Transformation().resize(autoPad(250, 250)).toString();
```
Comment (mckomo-cl) [inline __TESTS__/unit/actions/Resize/AutoPadResizeAction.test.ts:7]: I'm not sure if the syntax follows the convention. Currently, for `c_pad` it's

```ts
new CloudinaryImage("cld-sample.jpg").resize(pad().width(225).height(225));
```

Comment (magdakwiecien) [reply __TESTS__/unit/actions/Resize/AutoPadResizeAction.test.ts:7]: This is the alternative syntax, pad (and any other resize action) also supports this :) Take a look at [actions/resize.ts](https://github.com/cloudinary/js-transformation-builder-sdk/blob/master/src/actions/resize.ts#L94):

```
function pad(width?: string|number, height?: string|number) :ResizePadAction<CompassGravity> {
  return new ResizePadAction('pad', width, height);
}
```

Comment (mckomo-cl) [reply __TESTS__/unit/actions/Resize/AutoPadResizeAction.test.ts:7]: I would test two cases then ;)

**Review by mckomo-cl — APPROVED** [review]

File: __TESTS__/unit/actions/Resize/AutoPadResizeAction.test.ts (hunk 1 above)
Comment (magdakwiecien) [reply __TESTS__/unit/actions/Resize/AutoPadResizeAction.test.ts:7]: There are two cases 😄 please check the one below:
```
 it('Ensures it generates the right transformation using qualifiers', () => {
    const tx = new Transformation().resize(
      Resize.autoPad()
        .width(250)
        .height(250)
        .background(Background.color('red'))).toString();
```

(I basically copied this test suite from some other Resize action 😉 )

**Pull request merged**